src/tests/compiler/class/methodDeclarationTests.ts

```diff
@@ -202,4 +221,36 @@ describe(nameof(MethodDeclaration), () => {
             });
         });
     });
+
+    describe(nameof<MethodDeclaration>(d => d.getStructure), () => {
+        function doTest(code: string, expectedStructure: MethodDeclarationStructure) {
+            const {descendant, sourceFile} = getInfoFromTextWithDescendant<MethodDeclaration>(code, SyntaxKind.MethodDeclaration);
+            const structure = descendant.getStructure();
+            expect(Object.assign({}, structure, {parameters: undefined})).to.contain(Object.assign({}, expectedStructure, {parameters: undefined}));
+            expect(expectedStructure.parameters && expectedStructure.parameters.length).to.equals(structure.parameters && structure.parameters.length);
+            (expectedStructure.parameters || []).forEach((expectedParameter, i) => {
+                expect(structure.parameters![i], `${i}th parameter of method ${structure.name}`).to.contain(expectedParameter);
+            });
+            // and also recreate the AST using structure and compare getText() of generated code with original node's
+            const aux = sourceFile.addStatements("class __AuxClass{}");
+            const decl = aux[0];
+            if (!TypeGuards.isClassDeclaration(decl)) {
```

Comment:
Don't worry about adding this. It's obvious from above that it's a class so just assert it.